src/components/explore/AggregationsOverTimeGraph.svelte

```diff
@@ -143,6 +143,95 @@
     // Finally, set the flag to open the context menu.
     $showContextMenu = true; // eslint-disable-line no-unused-vars
   }
+
+  const getYTicks = (ranges) => {
+    if (
+      data[0].metric_type === 'histogram-linear' ||
+      yScaleType === 'scalePoint'
+    ) {
+      // when the range is too small we need to custom set it
+      if (ranges[ranges.length - 1] <= 5) return [0, 1, 2, 3, 4, 5];
+      if (yScaleType === 'scalePoint' && ranges.length < 5) return yValues;
+      return undefined;
+    }
+    if (
+      $store.activeBuckets.length &&
+      $store.proportionMetricType === 'proportions'
+    ) {
+      // when the percentage gets too small we need to manually set the ticks
+      if (ranges[ranges.length - 1] - ranges[0] < 0.05)
+        return [0, ranges[ranges.length - 1]];
+    }
+    if (
+      $store.activeBuckets.length &&
+      $store.proportionMetricType === 'counts'
+    ) {
+      if (ranges[ranges.length - 1] < 5) return [0, 5];
+    }
+    return undefined;
+  };
+
+  const getYDomain = (visiblePercentiles, buckets) => {
+    let yData = [];
+    let yDomainValues = [];
+
+    // linear and log graphs
+    if (
+      data[0].metric_type === 'histogram-linear' ||
+      yScaleType === 'scalePoint'
+    ) {
+      // eslint-disable-next-line no-restricted-syntax, guard-for-in
+      for (const p in visiblePercentiles) {
+        yData = [
+          ...yData,
+          ...data.map((arr) => arr.percentiles[visiblePercentiles[p]]),
+        ];
+      }
```

Comment:
This could be rewritten using `forEach` to satisfy eslint:
```js
      visiblePercentiles.forEach((p) => {
        yData = yData.concat(
          [...data.map((arr) => arr.percentiles[p])]
        );
      });
```